File: ddPoliglotV6.Data/Models/Lesson.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ddPoliglotV6.Data.Models
{
    public class Lesson
    {
        [Key]
        public int LessonID { get; set; }

        public int ParentID { get; set; } // Parent Lesson

        public int LanguageID { get; set; }

        [ForeignKey("LanguageID")]
        public Language Language { get; set; } // language to learn

        public int NativeLanguageID { get; set; } // native user language

        [MaxLength(250)]
        public string Name { get; set; }

        [MaxLength(250)]
        public string Title { get; set; }

        [MaxLength(3000)]
        public string Description { get; set; }

        public string Content { get; set; }

        public int Order { get; set; }

        [MaxLength(250)]
        public string Video1 { get; set; }

        [MaxLength(250)]
        public string Video2 { get; set; }

        [MaxLength(250)]
        public string Video3 { get; set; }

        [MaxLength(250)]
        public string Video4 { get; set; }

        [MaxLength(250)]
        public string Video5 { get; set; }

        [MaxLength(250)]
        public string Audio1 { get; set; }

        [MaxLength(250)]
        public string Audio2 { get; set; }

        [MaxLength(250)]
        public string Audio3 { get; set; }

        [MaxLength(250)]
        public string Audio4 { get; set; }

        [MaxLength(250)]
        public string Audio5 { get; set; }

        [MaxLength(250)]
        public string Image1 { get; set; }

        [MaxLength(250)]
        public string Image2 { get; set; }

        [MaxLength(250)]
        public string Image3 { get; set; }

        [MaxLength(250)]
        public string Image4 { get; set; }

        [MaxLength(250)]
        public string Image5 { get; set; }
        [MaxLength(3000)]
        public string Description1 { get; set; }

        [MaxLength(3000)]
        public string Description2 { get; set; }

        [MaxLength(3000)]
        public string Description3 { get; set; }

        [MaxLength(3000)]
        public string Description4 { get; set; }

        [MaxLength(3000)]
        public string Description5 { get; set; }

        public DateTime Modified { get; set; }

        [MaxLength(20)]
        public string State { get; set; }

        [MaxLength(20)]
        public string PageName { get; set; }
        public int ArticleByParamID { get; set; }

        [MaxLength(500)]
        public string PageMetaTitle { get; set; }

        [MaxLength(1000)]
        public string PageMetaDescription { get; set; }

        [NotMapped]
        public ArticleByParam ArticleByParam { get; set; }
    }
}

File: ddpoliglot_v6_flutter/lib/models/data/Lesson.dart
// ignore_for_file: file_names
import './Language.dart';
import 'article_by_param.dart';

import 'package:json_annotation/json_annotation.dart';

part 'Lesson.g.dart';

@JsonSerializable()
class Lesson {
  int lessonID;
  int parentID; // Parent Lesson
  int languageID;
  Language language; // language to learn
  int nativeLanguageID; // native user language
  String name;
  String title;
  String description;
  String content;
  int order;
  String video1;
  String video2;
  String video3;
  String video4;
  String video5;
  String audio1;
  String audio2;
  String audio3;
  String audio4;
  String audio5;
  String description1;
  String description2;
  String description3;
  String description4;
  String description5;
  DateTime modified;
  String state;
  String pageName;
  int articleByParamID;
  ArticleByParam articleByParam;

  Lesson(
    this.lessonID,
    this.parentID, // Parent Lesson
    this.languageID,
    this.language, // language to learn
    this.nativeLanguageID, // native user language
    this.name,
    this.title,
    this.description,
    this.content,
    this.order,
    this.video1,
    this.video2,
    this.video3,
    this.video4,
    this.video5,
    this.audio1,
    this.audio2,
    this.audio3,
    this.audio4,
    this.audio5,
    this.description1,
    this.description2,
    this.description3,
    this.description4,
    this.description5,
    this.modified,
    this.state,
    this.pageName,
    this.articleByParamID,
    this.articleByParam,
  );

  factory Lesson.fromJson(Map<String, dynamic> data) => _$LessonFromJson(data);

  Map<String, dynamic> toJson() => _$LessonToJson(this);
}
